modules/Microsoft.App/managedEnvironments/deploy.bicep

```diff
@@ -0,0 +1,171 @@
+param vnetOutboundSettings object = {}
+
+@description('Optional. IP range in CIDR notation that can be reserved for environment infrastructure IP addresses. Must not overlap with any other provided IP ranges.')
+param platformReservedCidr string = ''
+
+@description('Optional. An IP address from the IP range defined by platformReservedCidr that will be reserved for the internal DNS server.')
+param platformReservedDnsIP string = ''
+
+@description('Optional. Resource ID of a subnet that Container App containers are injected into. This subnet must be in the same VNET as the subnet defined in infrastructureSubnetId. Must not overlap with any other provided IP ranges.')
+param runtimeSubnetId string = ''
+
+@description('Optional. Whether or not this Managed Environment is zone-redundant.')
+param zoneRedundant bool = false
+
+@description('Optional. Password of the certificate used by the custom domain.')
+@secure()
+param certificatePassword string = ''
+
+@description('Optional. Certificate to use for the custom domain. PFX or PEM.')
+@secure()
+param certificateValue string = ''
+
+@description('Optional. DNS suffix for the environment domain.')
+param dnsSuffix string = ''
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Workload profiles configured for the Managed Environment.')
+param workloadProfiles array = []
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2022-09-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2022-10-01' existing = {
+  name: last(split(logAnalyticsWorkspaceResourceId, '/'))!
+  scope: resourceGroup(split(logAnalyticsWorkspaceResourceId, '/')[2], split(logAnalyticsWorkspaceResourceId, '/')[4])
+}
+
+resource managedEnvironment 'Microsoft.App/managedEnvironments@2022-06-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: skuName
+  }
+  properties: {
+    appLogsConfiguration: {
+      destination: logsDestination
+      logAnalyticsConfiguration: {
+        customerId: logAnalyticsWorkspace.properties.customerId
+        sharedKey: logAnalyticsWorkspace.listKeys().primarySharedKey
+      }
+    }
+    daprAIConnectionString: daprAIConnectionString
+    daprAIInstrumentationKey: daprAIInstrumentationKey
+    customDomainConfiguration: {
+      certificatePassword: certificatePassword
+      certificateValue: !empty(certificateValue) ? certificateValue : null
+      dnsSuffix: dnsSuffix
+    }
+    vnetConfiguration: {
```

Comment:
needs a condition if 'internal' is set to `true`, else `null`

Reply:
so the entire `vnetConfiguration` block should only be used if `internal` is `true`, or specific settings? Just want to be sure

Reply:
```suggestion
    vnetConfiguration: {
      internal: internal
      infrastructureSubnetId: !empty(infrastructureSubnetId) && internal == true ? infrastructureSubnetId : null
      dockerBridgeCidr: !empty(infrastructureSubnetId) && internal == true ? dockerBridgeCidr : null
      platformReservedCidr: !empty(infrastructureSubnetId) && internal == true ? platformReservedCidr : null
      platformReservedDnsIP: !empty(infrastructureSubnetId) && internal == true ? platformReservedDnsIP : null
    }
```